backend/routes/uploadRoutes.js

```diff
@@ -15,26 +16,33 @@ const storage = multer.diskStorage({
   },
 });
 
-function checkFileType(file, cb) {
-  const filetypes = /jpg|jpeg|png/;
+function fileFilter(req, file, cb) {
```

Comment:
I've renamed the function to fileFilter for convenience, since multer looks for a key of fileFilter when calling the multer function below. docs: [multer options](https://github.com/expressjs/multer#multeropts)